### DISCUSSION TITLE
Why is my page load over 1.5 second with that little data?

### DISCUSSION THREAD

**Discussion by SelfDevTV** [prisma/prisma#4878] [Q&A] [answered]
Hi guys,

I'm building a reddit clone right now with `Prisma` + `AWS` (as PostgreSQL Hoster) + `Next.js`. I'm fetching a subreddit with all details on it (mainly posts and subscribed users) in `getServerSideProps`. I make a fetch request to my api and in my api I make a call to my prisma database (hosted on Amazon AWS - PostgreSQL). 

Here is the working live example (just select a subreddit in the dropdown). Next.js has some posts, learnprogramming hasnt: https://reddit-clone-teal.vercel.app/

Is that slow / fast to you? For me it's a bit slow with that little data (Vercel Analytics = 3 seconds) 😢 

**Here is the api:**

```
const handler = async (req: NextApiRequest, res: NextApiResponse) => {
  try {
    const sub = await prisma.subreddit.findUnique({
      where: { name: String(req.query.name) },
      include: {
        posts: {
          include: { subreddit: true, user: true },
        },
        joinedUsers: true
      },
    });

    console.log("here is the sub", sub);

    if (!sub) {
      return res.status(500).json({ error: "No sub was found with this name" });
    }

    res.json(sub);
  } catch (e) {
    res.json(e);
  }
};
```

So nothing too crazy right (it's 3 posts total)? Or am I including too much in the call?
Anyway when I transition to a subreddit and this function runs (there are currently 3 dummy posts in the db) it takes sometimes 1.5 - 2 seconds after the page has loaded in.
With such little data this should be much less loading time am I right?

I'm using `swr` and here is how a subreddit page looks like:

**First the `getServerSideProps` function:**

```
export async function getServerSideProps(ctx) {
  const url = `${process.env.NEXTAUTH_URL}/api/subreddit/findSubreddit?name=${ctx.query.sub}`;
  const fullSub = await fetchData(url);

  return {
    props: {
      fullSub,
    },
  };
}
```

**And then on the page itself:**

```
const { data: fullSub, error } = useSWR(
    `/api/subreddit/findSubreddit?name=${sub}`,
    fetchData,
    {
      initialData: props.fullSub,
    }
  );
```

**Offtopic (Next.js specific)**
It also kinda loads it everytime (going from one subreddit to another), although I'm caching it with `swr`. I think I'm doing something wrong. It's kinda faster then the first transition from subreddit A to subreddit B but shouldn't it be instant when nothing has changed? Isn't that the advantage of using `swr` or am I missing something again? 😄 I think that's because ssr doesn't cache because it's on the server. But's thats offtopic. 

**Here is the log from prisma when I hit this api:**

>prisma:query SELECT "public"."Subreddit"."id", "public"."Subreddit"."name", "public"."Subreddit"."displayName", >"public"."Subreddit"."infoBoxText", "public"."Subreddit"."createdAt" FROM "public"."Subreddit" WHERE >"public"."Subreddit"."name" = $1 LIMIT $2 OFFSET $3
>prisma:query SELECT "public"."Post"."id", "public"."Post"."title", "public"."Post"."body", "public"."Post"."subredditId", >"public"."Post"."userId", "public"."Post"."createdAt", "public"."Post"."updatedAt" FROM "public"."Post" WHERE >"public"."Post"."subredditId" IN ($1) OFFSET $2
>prisma:query SELECT "public"."users"."id", "public"."users"."name", "public"."users"."email", "public"."users"."email_verified", >"public"."users"."image", "public"."users"."created_at", "public"."users"."updated_at" FROM "public"."users" WHERE >"public"."users"."id" IN ($1) OFFSET $2
>prisma:query SELECT "public"."_SubredditToUser"."A", "public"."_SubredditToUser"."B" FROM "public"."_SubredditToUser" WHERE >"public"."_SubredditToUser"."A" IN ($1)
>prisma:query SELECT "public"."users"."id", "public"."users"."name", "public"."users"."email", "public"."users"."email_verified", >"public"."users"."image", "public"."users"."created_at", "public"."users"."updated_at" FROM "public"."users" WHERE >"public"."users"."id" IN ($1) OFFSET $2

Here a screenshot from Vercel Analytic:

![image](https://user-images.githubusercontent.com/48333000/103637252-7b050480-4f4b-11eb-9d44-a597607fc3bb.png)

Default Load time from Postman (Get to `https://reddit-clone-git-master.selfdevtv.vercel.app/api/subreddit/findSubreddit?name=nextjs`):

![image](https://user-images.githubusercontent.com/48333000/103584359-6db33000-4ee1-11eb-9430-00a783dbbfb6.png)
I know that's not too bad, but again for this  tiny bit of data I would expect a response in 300 - 600ms or so.

Can you figure it out? Any ideas why this might long load and not cache? The db is hosted in frankfurt (on AWS) don't know if that's important to you. But im located in Austria and my ping to Frankfurt is 45ms. 

Full link to my repo: https://github.com/SelfDevTV/reddit-clone

Thank you so much!
_**PS: Prisma rocks!**_

**Comment by nikolasburk** [comment, marked as answer]
Hey @SelfDevTV, thanks a lot for raising this! This sounds indeed like something our engineering team should look into, do you mind [reposting this as an actual issue](https://github.com/prisma/prisma/issues/new) so that the engineers can look into it? (We use discussions for questions and support while issues are investigated by the engineering team directly).

**Reply by SelfDevTV** [reply]
Hi @nikolasburk 🤟 

Thx for answering. Yes sure I will be creating a new issue. I also posted something in this issue today:

https://github.com/prisma/prisma/issues/2481

Thx again,
Greetings from Austria :) 
PS: You guys rock!

**Reply by SelfDevTV** [reply]
@nikolasburk I opened it: https://github.com/prisma/prisma/issues/4884
I hope you guys can figure it out 😃

**Reply by nikolasburk** [reply]
Hey @SelfDevTV, so glad to see that this has been resolved 🙂 Seems like there might a be a couple of interesting lessons in there that might be relevant to other folks (e.g. making sure DB and API server are close to each other, usage of free tier DBs and pgBouncer, ...). I don't know if you're into blogging, but I think it could be super interesting for other people to read a "summary" of the issues you faced and the resolutions for them, so this might be a great topic to write about! (If you don't have a blog yet, communities like [dev.to](https://dev.to/) or [Hashnode](https://hashnode.com/)) make it really easy to start writing without the need to set up an entire blog first). I'd be happy to help with this where I can, feel free to ping me on the [Prisma Slack](https://slack.prisma.io) or shoot an email to burk@prisma.io if you ever decide to work on this 🙌

**Reply by SelfDevTV** [reply]
Hi @nikolasburk 

Good idea! I did just that: https://dev.to/selfdevtv/what-i-learned-by-using-prisma-with-postgresql-2b6c

I hope you all have a good read and if you find any errors (which I think you will) please let me know. This technology especially `SQL` and `PostgreSQL` are pretty new to me, so I hope I understood everythong so far correctly. 

Thank you!

**Reply by nikolasburk** [reply]
This is amazing, thank you so much 😄 I'm sure this will help lots of other people!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
